### PR TITLE
get_db_matching_location: Dont remove NaN, don't catch exceptions

### PR DESCRIPTION
**Pull request by elazarg**
Using `fillna` was a mistake. <s>: None cannot be put in the DataFrame.</s> So an exception was thrown, and got caught by `except Exception`, without logging the actual reason for the failure. 

This PR <s>simply remove both operations.</s> reverts the use of `fillna` to `replace`, and removes the `try-except` block.

### CODE REVIEW

File: anyway/parsers/location_extraction.py [hunk 1]
```diff
@@ -41,64 +41,50 @@ def get_db_matching_location(db, latitude, longitude, resolution, road_no=None):
...
+            road_no is not None
+            and road_no > 0
+            and ("road1" in relevant_fields or "road2" in relevant_fields)
+        ):
+            markers = markers.loc[(markers["road1"] == road_no) | (markers["road2"] == road_no)]
+        for field in relevant_fields:
+            if field == "road1":
+                markers = markers.loc[markers[field].notnull()]
+                markers = markers.loc[markers[field] > 0]
+            elif field in ["region_hebrew", "district_hebrew", "yishuv_name", "street1_hebrew"]:
```
Comment (elazarg) [inline anyway/parsers/location_extraction.py:63]: Here's a small change unrelated to the core issue - simple refactoring.

File: anyway/parsers/location_extraction.py [hunk 2]
```diff
@@ -41,64 +41,50 @@ def get_db_matching_location(db, latitude, longitude, resolution, road_no=None):
...
-        )
-        for field in relevant_fields:
-            loc = most_fit_loc[field]
-            if loc not in [None, "", "nan"]:
-                if not (isinstance(loc, np.float64) and np.isnan(loc)):
-                    final_loc[field] = loc
+    # CREATE DISTANCE FIELD
+    markers["dist_point"] = markers.apply(
+        lambda x: geod.Inverse(latitude, longitude, x["latitude"], x["longitude"])["s12"], axis=1,
+    )
```
Comment (elazarg) [inline anyway/parsers/location_extraction.py:77]: Here we tried to avoid having a NaN value, but what of it?
If we must, we can change the lambda to not return a NaN.

File: anyway/parsers/location_extraction.py [hunk 3]
```diff
@@ -41,64 +41,50 @@ def get_db_matching_location(db, latitude, longitude, resolution, road_no=None):
...
-        most_fit_loc = (
-            markers.loc[markers["dist_point"] == markers["dist_point"].min()].iloc[0].to_dict()
-        )
-        for field in relevant_fields:
-            loc = most_fit_loc[field]
-            if loc not in [None, "", "nan"]:
-                if not (isinstance(loc, np.float64) and np.isnan(loc)):
-                    final_loc[field] = loc
+    # CREATE DISTANCE FIELD
+    markers["dist_point"] = markers.apply(
```
Comment (atalyaalon) [inline anyway/parsers/location_extraction.py:75]: Same here, identation is changed

File: anyway/parsers/location_extraction.py [hunk 4]
```diff
@@ -41,64 +41,50 @@ def get_db_matching_location(db, latitude, longitude, resolution, road_no=None):
...
+                markers = markers.loc[markers[field] != ""]
+    if markers.count()[0] == 0:
+        markers = markers_orig
 
-        # FILTER BY GEOHASH
-        curr_geohash = geohash.encode(latitude, longitude, precision=4)
-        if markers.loc[markers["geohash"] == curr_geohash].count()[0] > 0:
-            markers = markers.loc[markers["geohash"] == curr_geohash].copy()
+    # FILTER BY GEOHASH
+    curr_geohash = geohash.encode(latitude, longitude, precision=4)
```
Comment (atalyaalon) [inline anyway/parsers/location_extraction.py:70]: why did you change the indentation?

**Comment by atalyaalon** [comment]
@elazarg are you sure that None cannot be put in a dataframe?
From what I recall we had issues with np.nan inserted into the DB, that's why we had [such line](https://github.com/hasadna/anyway/blob/1585772bda6cb3cdeb9411b319024c5700baab31/anyway/parsers/news_flash_parser.py#L113) that makes sure no np.nan is inserted to the DB, only None, for example, in the past

File: anyway/parsers/location_extraction.py (hunk 4 above)
Comment (elazarg) [reply anyway/parsers/location_extraction.py:70]: It is no longer inside an exception clause.

**Comment by elazarg** [comment]
Pandas does not recognize `df.fillna(None)` as having an argument at all. However, regarding the None, I had a discussion with @Mano3 about `x.replace({nan: None})` not having any effect - but now that I re-checked it, it turns out I had a typo... :( 
So yeah, we may have `x.replace({nan: None})`. I will fix that momentarily.

@Mano3 sorry for misleading you here! `x.replace({nan, None})` indeed does not work, and for a good reason.

File: anyway/parsers/location_extraction.py (hunk 2 above)
Comment (elazarg) [reply anyway/parsers/location_extraction.py:77]: I will fix that.

**Comment by atalyaalon** [comment]
@elazarg can I squash and merge this one? Should the issue be fixed?
